### ISSUE TITLE
org.json.JSONException: JSONObject["ios"] not found.

### ISSUE THREAD

**Issue by zarashima** [AppiumTestDistribution/AppiumTestDistribution#513]
- ATD version: 8f1e2e1156 from this PR: https://github.com/AppiumTestDistribution/AppiumTestDistribution/issues/493
- Capabilities:
```
{
  "android": {
    "app": {
      "local": "myApp.apk"
    },
    "platformName": "Android",
    "automationName":"UIAutomator2",
    "deviceName": "android",
    "noSign": true,
    "noReset": true,
    "autoGrantPermissions": true,
    "deviceReadyTimeout": 10,
    "ignoreUnimportantViews": true,
    "androidDeviceReadyTimeout": 300
  },
  "iOS": {
    "platformName": "iOS",
    "app": {
      "device": "myApp.ipa"
    },
    "automationName": "XCUITest",
    "deviceName": "iPhone",
    "wdaLaunchTimeout": 80000,
    "wdaConnectionTimeout": 80000,
    "preventWDAAttachments": true,
    "clearSystemFiles": true,
    "shouldUseSingletonTestManager": false,
    "simpleIsVisibleCheck": true,
    "maxTypingFrequency": 10,
    "xcodeOrgId": "my id",
    "xcodeSigningId": "iPhone Developer",
    "wdaStartupRetries": 4,
    "autoAcceptAlerts" : true,
    "useNewWDA": true,
    "noReset": true
  },
  "hostMachines": [
    {
      "machineIP": "127.0.0.1",
      "appiumServerPath": "/usr/local/lib/node_modules/appium/build/lib/main.js"
    }
  ]
}
```

Upon executing this ATD version for iOS, I observe the following error:
```INFO: Allocated Device com.appium.manager.AppiumDevice@4dcfc3ff for Execution
org.json.JSONException: JSONObject["ios"] not found.
        at org.json.JSONObject.get(JSONObject.java:566)
        at org.json.JSONObject.getJSONObject(JSONObject.java:778)
        at com.appium.capabilities.DesiredCapabilityBuilder.desiredCapabilityForLocalAndRemoteATD(DesiredCapabilityBuilder.java:107)
        at com.appium.capabilities.DesiredCapabilityBuilder.buildDesiredCapability(DesiredCapabilityBuilder.java:54)
        at com.appium.manager.AppiumDriverManager.buildDesiredCapabilities(AppiumDriverManager.java:116)
        at com.appium.manager.AppiumDriverManager.startAppiumDriverInstance(AppiumDriverManager.java:93)
        at com.appium.manager.AppiumParallelTestListener.onTestStart(AppiumParallelTestListener.java:156)
        at org.testng.internal.Invoker.runTestListeners(Invoker.java:1739)
        at org.testng.internal.Invoker.runTestListeners(Invoker.java:1714)
        at org.testng.internal.Invoker.invokeMethod(Invoker.java:640)
        at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:869)
        at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1193)
        at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:126)
        at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:109)
        at org.testng.TestRunner.privateRun(TestRunner.java:744)
        at org.testng.TestRunner.run(TestRunner.java:602)
        at org.testng.SuiteRunner.runTest(SuiteRunner.java:380)
        at org.testng.SuiteRunner.access$000(SuiteRunner.java:39)
        at org.testng.SuiteRunner$SuiteWorker.run(SuiteRunner.java:414)
        at org.testng.internal.thread.ThreadUtil$1.call(ThreadUtil.java:52)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
```
However when I revert ATD version to be 10.0.1, then iOS devices execution work fine. So is there something wrong with my capabilities on iOS device? Or is it an issue from that specific version of ATD: 8f1e2e1156

Regards

**Comment by saikrishna321** [comment]
@zarashima Looks like something is broken at the DeviceManager from iOS side. Can you please confirm which version of xcode and deviceOS version?

As workaround can you change "iOS" to "ios" in your capability.json 

```
"ios": {
    "platformName": "iOS",
    "app": {
      "device": "myApp.ipa"
    },
    "automationName": "XCUITest",
    "deviceName": "iPhone",
    "wdaLaunchTimeout": 80000,
    "wdaConnectionTimeout": 80000,
    "preventWDAAttachments": true,
    "clearSystemFiles": true,
    "shouldUseSingletonTestManager": false,
    "simpleIsVisibleCheck": true,
    "maxTypingFrequency": 10,
    "xcodeOrgId": "my id",
    "xcodeSigningId": "iPhone Developer",
    "wdaStartupRetries": 4,
    "autoAcceptAlerts" : true,
    "useNewWDA": true,
    "noReset": true
  }

```

**Comment by zarashima** [comment]
Here is my version:
- Xcode: 10.2.1
- iOS: 12.4

I've changed from 'iOS' to 'ios', observe the similar issue:

```
Jul 08, 2019 3:50:10 PM com.appium.schema.CapabilitySchemaValidator validateCapabilitySchema
INFO: required key [iOS] not found
[ERROR] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.373 s <<< FAILURE! - in com.appium.tests.runner.runner
[ERROR] testRunner(com.appium.tests.runner.runner)  Time elapsed: 0.173 s  <<< FAILURE!
org.everit.json.schema.ValidationException: #: Capability json provided is missing the above schema
```

**Comment by zarashima** [comment]
Any other info? :)

**Comment by saikrishna321** [comment]
Working on it today!! Will keep the ticket updated

**Comment by saikrishna321** [comment]
@zarashima I'm unable to reproduce this. Can you please join the slack group 

https://join.slack.com/t/appiumtestparallel/shared_invite/enQtMjIxMjIwMzMyMTAyLTI2NzAyM2E5ZjJkZmJkYzg2MzQxZjBkZjkzYTliOWEzNmQyYWY2MGEyMGY2MGY4MDI2YjU5ZDBmZjRlNGJmZTU

**Comment by zarashima** [comment]
Will do later, but first are you using this ATD version: 8f1e2e1? And can you post for me your current working desired capabilities?

**Comment by saikrishna321** [comment]
https://github.com/AppiumTestDistribution/AppiumTestDistribution/blob/master/caps/capabilities.json

This issue seems like a real device problem and I don't hold one to debug.

**Comment by zarashima** [comment]
OK wait let me try once again from my side first

**Comment by saikrishna321** [comment]
@zarashima Can you try with the below version 

```
	<dependency>
	    <groupId>com.github.AppiumTestDistribution</groupId>
	    <artifactId>AppiumTestDistribution</artifactId>
	    <version>c5c657c32f</version>
	</dependency>
```

**Comment by zarashima** [comment]
@saikrishna321 : Sure let me try it

**Comment by zarashima** [comment]
I've confirmed it's working for iOS. Let me try again to see if adb log files are generated correctly for more than one device.

**Comment by saikrishna321** [comment]
Thanks for confirming. Closing as fixed. Please raise another issue is adb log issue exists
